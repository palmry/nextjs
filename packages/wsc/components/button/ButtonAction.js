import React from "react"
import styled from "styled-components"
import PropTypes from "prop-types"
import { COLORS, FONT_FAMILIES, withButton } from "../../utils/styles"
import { ReactComponent as IconNavRight } from "../../statics/images/icon-nav-right.svg"

const Button = styled.button.attrs({
  className: "buttonAction",
})`
  ${withButton}
  color: white;
  height: var(--buttonAction_Height, 40px);
  width: 100%;
  max-width: 368px;
  border: none;
  cursor: pointer;
  font-size: 0.93rem;
  font-weight: var(--buttonAction_FontWeight, 600);
  font-stretch: normal;
  font-style: normal;
  vertical-align: bottom;
  line-height: 1.73;
  font-family: var(--buttonAction_FontFamily, ${FONT_FAMILIES.SANSSERIF});
  ${(props) =>
    !props.active &&
    `
    pointer-events: none;
    color: var(--buttonAction_Disable_FontColor, white);
    background-color: var(--buttonAction_Disable_BackgroundColor, ${COLORS.DISABLE});
    svg { fill: var(--buttonAction_Disable_FontColor, ${COLORS.WHITE}); }
    cursor: auto;
  `}
  outline: none;
  -webkit-user-select: none; /* Disable hilighting Chrome all / Safari all */
  -moz-user-select: none; /* Disble hilighting Firefox all */
  -ms-user-select: none; /* Disable hilighting IE 10+ */
  user-select: none; /* Disable hilighting Likely future *
`
const ScImageBox = styled.div`
  display: inline;
  margin-left: 6px;
  svg {
    height: 0.46rem;
    width: 0.46rem;
    vertical-align: middle;
  }
`
export const ButtonAction = (props) => {
  const click = (e) => {
    props.action()
  }
  ButtonAction.propTypes = {
    text: PropTypes.string.isRequired,
    active: PropTypes.bool.isRequired,
    action: PropTypes.func.isRequired,
    isWithArrow: PropTypes.bool,
  }
  ButtonAction.defaultProps = {
    isWithArrow: false,
  }
  return (
    <Button active={props.active} onClick={click}>
      {props.text}
      {props.isWithArrow && (
        <ScImageBox>
          <IconNavRight />
        </ScImageBox>
      )}
    </Button>
  )
}
