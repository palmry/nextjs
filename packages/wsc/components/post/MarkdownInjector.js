import React, { Fragment } from 'react'
import ContentMarkdown from './ContentMarkdown'
// import EmbedImage from './EmbedImage'
import PropTypes from 'prop-types'
// import { JSDOM } from 'jsdom'

let preProcessed = null
// const EMBED_TAG_REGEXP = /<div.+?>*<\/div>|<iframe.+?><\/iframe>|<blockquote.+?(twitter|instagram|tiktok).+?>*<\/blockquote>|<a.+?>Embed*<\/a>|<script.+?><\/script>|<script.+?\/>/gs

// placeholder for markdown in componentList
class MarkdownPlaceholder {
  constructor(markdown, key) {
    this.markdown = markdown
    this.key = key
  }
}

// Fixes formatting of markdown so headings, images, etc have appropriate line spacing
const getParagraphsFromMarkdown = markdown => {
  const lines = markdown.split('\n')
  const paragraphs = []
  let paragraph = ''
  let paragraphType = 't'
  const orderedListRegex = /^[0-9]+\. /

  const addNewParagraph = newParagraph => {
    //close previous paragraph
    if (paragraph !== '') {
      paragraphs.push(paragraph + '\n')
      paragraph = ''
      paragraphType = 't'
    }

    //also add new one if provided
    if (newParagraph) paragraphs.push(newParagraph + '\n\n')
  }

  // process each line
  lines.forEach(line => {
    //blank line denotes a new paragraph
    if (line.trim() === '') {
      addNewParagraph()
    }
    // headings
    else if (line[0] === '#') {
      addNewParagraph(line)
    }
    // images
    else if (line.length > 2 && line[0] === '!' && line[1] === '[') {
      addNewParagraph(line)
    }
    // unordered lists
    else if (line[0] === '*' || line[0] === '-' || line[0] === '+') {
      const listType = line[0]
      //if we aren't building the same list type as this one, close current paragraph
      if (paragraphType !== listType) {
        addNewParagraph()
        paragraphType = listType
      }
      paragraph += line + '\n'
    }
    // ordered lists
    else if (orderedListRegex.test(line)) {
      if (paragraphType !== 'ol') {
        addNewParagraph()
        paragraphType = 'ol'
      }
      paragraph += line + '\n'
    }
    // all others (basic text)
    else {
      // if we were building a non-basic-text paragraph before, close it since we're at text again
      if (paragraphType !== 't') addNewParagraph()
      // append to last line
      paragraph += line + '\n'
    }
  })
  paragraphs.push(paragraph)
  return paragraphs
}

export const createMarkdownPlaceHolder = (markdown, key) => {
  return [new MarkdownPlaceholder(markdown, key)]

  /*
  We will enable this once we fixed twitter quote issue.
  let regExpExecArray
  let markdownPlaceholders = []
  let keyCounter = 0
  try {
    let element = new JSDOM(markdown)
    // cleanup html
    const tempMarkdown = element.window.document.body.innerHTML
    let markdownBuffer = tempMarkdown

    // find embed tags then rebuild them by EmbedImage component
    while ((regExpExecArray = EMBED_TAG_REGEXP.exec(tempMarkdown)) !== null) {
      let matchedEmbedTag = regExpExecArray[0]
      const tagIndex = markdownBuffer.indexOf(matchedEmbedTag)
      const markdownContent = markdownBuffer.substring(0, tagIndex)
      let isScript = false
      // There is a limitation in regex. It cannot determine where the closed element is if there are many child.
      // So, we use jsdom to query element to ensure we pick all child in target element.
      if (matchedEmbedTag.indexOf('<div') === 0) {
        element = new JSDOM(markdownBuffer)
        if (matchedEmbedTag.includes('class="tumblr-post"')) {
          matchedEmbedTag = element.window.document.querySelector(`[class="tumblr-post"]`).outerHTML
        } else if (matchedEmbedTag.includes('itemtype="https://schema.org/VideoObject"')) {
          matchedEmbedTag = element.window.document.querySelector(
            `[itemtype="https://schema.org/VideoObject"]`
          ).outerHTML
        } else {
          continue
        }
      } else if (matchedEmbedTag.indexOf('<script') === 0) {
        let lastComponent = markdownPlaceholders[markdownPlaceholders.length - 1]
        if (lastComponent instanceof MarkdownPlaceholder) {
          lastComponent.markdown += matchedEmbedTag
        } else {
          markdownPlaceholders[markdownPlaceholders.length - 1] = (
            <EmbedImage htmlCode={lastComponent.props.htmlCode + matchedEmbedTag}></EmbedImage>
          )
        }
        isScript = true
      }

      // remove used content from buffer
      markdownBuffer = markdownBuffer.replace(markdownContent + matchedEmbedTag, '')

      if (isScript) continue

      // put normal markdown
      markdownPlaceholders.push(
        new MarkdownPlaceholder(markdownContent, `${key}-markdown-${keyCounter}`)
      )

      // put embed markdown
      markdownPlaceholders.push(
        <EmbedImage
          htmlCode={matchedEmbedTag}
          key={`${key}-embed-image-${keyCounter}`}
        ></EmbedImage>
      )

      keyCounter++
    }

    // put rest content
    if (markdownBuffer !== '') {
      markdownPlaceholders.push(
        new MarkdownPlaceholder(markdownBuffer, `${key}-markdown-${keyCounter}`)
      )
    }
  } catch (e) {
    // Do not convert any embed image if there is an error.
    console.error('Could not process the markdown due to error. ', e)
    markdownPlaceholders = [new MarkdownPlaceholder(markdown, `${key}-markdown-${keyCounter}`)]
  }
  return markdownPlaceholders*/
}

export const processMarkdown = (injectors, markdown, injectorOptions = {}) => {
  if (markdown === null || markdown === '') return

  const paragraphs = getParagraphsFromMarkdown(markdown)

  //build array of components
  let mdBuffer = ''
  let components = []
  let componentCounter = 0
  paragraphs.forEach((paragraph, paragraphIndex, paragraphs) => {
    let beforeList = []
    let afterList = []

    if (paragraph.trim() !== '') {
      // run injectors on the paragraph
      injectors.forEach(injector => {
        const result = injector(paragraphIndex, paragraph, injectorOptions)
        if (result) {
          if (result.before) beforeList = beforeList.concat(result.before)
          if (result.after) afterList = afterList.concat(result.after)
        }
      })

      // inject components before paragraph
      if (beforeList.length > 0) {
        if (mdBuffer !== '') {
          components = components.concat(createMarkdownPlaceHolder(mdBuffer, componentCounter++))
          mdBuffer = ''
        }

        components = components.concat(
          beforeList.map(comp => <Fragment key={componentCounter++}>{comp}</Fragment>) //items require key
        )
      }
      mdBuffer += paragraph

      //inject components after paragraph
      if (afterList.length > 0) {
        if (mdBuffer !== '') {
          components = components.concat(createMarkdownPlaceHolder(mdBuffer, componentCounter++))
          mdBuffer = ''
        }

        components = components.concat(
          afterList.map(comp => <Fragment key={componentCounter++}>{comp}</Fragment>) //items require key
        )
      }
    }
  })

  if (mdBuffer !== '') {
    components = components.concat(createMarkdownPlaceHolder(mdBuffer, componentCounter))
  }

  preProcessed = {
    markdown,
    components,
  }
  return components
}

const MarkdownInjector = props => {
  const { componentInjectors, markdown, injectorOptions } = props
  let injectors = componentInjectors
  if (markdown === null || markdown === '') return null

  if (!injectors) {
    return <ContentMarkdown {...props} content={markdown} />
  }
  // can be array or single function
  if (!Array.isArray(injectors)) injectors = [injectors]

  let components
  if (preProcessed != null && preProcessed.markdown === markdown) {
    components = preProcessed.components
  } else {
    components = processMarkdown(injectors, markdown, injectorOptions)
  }

  components.forEach((component, index) => {
    if (component instanceof MarkdownPlaceholder) {
      components[index] = (
        <ContentMarkdown {...props} content={component.markdown} key={component.key} />
      )
    }
  })

  return <>{components}</>
}
MarkdownInjector.defaultProps = {
  markdown: '',
  injectorOptions: {},
}
MarkdownInjector.propTypes = {
  markdown: PropTypes.string,
  componentInjectors: PropTypes.array.isRequired,
  injectorOptions: PropTypes.shape({}),
}

export default MarkdownInjector
