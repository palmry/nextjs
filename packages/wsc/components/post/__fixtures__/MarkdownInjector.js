export const DATA = {
  markdownData: `**1st paragraph starts**[Wayfair's \"Way Day\" sale](https://www.wayfair.com/daily-sales/way-day) totally beats them all\n\n**2nd paragraph starts**`,
  embedCodeData: `## __*Test*__ :bold + Italic : H2

  ### Italic : H3
  *This text will be italic*
  _This will also be italic_
  
  ### Bold : H3
  **This text will be bold** , 
  __This will also be bold__,
  _You **can** combine them_
  
  ### Strikethrough : H3
  ~~This should be italic~~, 
  __~~This should be italic~~__ ,  
  *~~This should be italic~~*,
  *__~~This should be italic~~__*
  
  ## NORMAL
  Mother's Day is a source of joy for most moms. A full day where you get what you want and your family waits on you hand and foot, right? Well, not always! Moms can sometimes get the short end of the stick — even on a day specifically made for us.
  
  ## Testing : H2 : WYSIWYG
  ## *"This should be italic h2"* 
  ### *"This should be italic h3"*
  
  Normal : 
  text *"This should be italic"* : *"Test55565 : WYSIWYG"*
  *Text Italic* 
  Yes, we are moms __(awesome, exhausted and busy)__ __*"This should be italic"*__, but we are also women (with jobs in and out of the home; *with partners, interests and friends).*
  
  ~~"This should be italic"~~
  
  ###   Unordered
  Version1 : 
  * Item 1
  * Item 2
  
  Version2 : 
  - Test1
  - Test2
  
  ###   Order List
  1. Number 
  2. No
  3. Sum
  
  ### Images
    ![HB](//images.ctfassets.net/9l3tjzgyn9gr/7uLuGY7SJq9dFfrfnWQTKz/505f757e2a61726bb1c73e7d46ac6ac5/HB_LogoWithBackground-01.png)
  
  ### Links
  1. http://github.com - automatic!
  
  2. [GitHub](http://github.com)
  
  ### Blockquotes
  As Kanye West said:
  
  > We're living the future so
  > the present is our past.
  
  ### Inline code
  I think you should use an
   <addr> element here instead.
  
  😇 😀
  
  ### ฺBlockquote
  - Instragram
  <blockquote class="instagram-media" data-instgrm-permalink="https://www.instagram.com/p/B7EbX3hnnbu/?utm_source=ig_embed&utm_campaign=loading" data-instgrm-version="12" style=" background:#FFF; border:0; border-radius:3px; box-shadow:0 0 1px 0 rgba(0,0,0,0.5),0 1px 10px 0 rgba(0,0,0,0.15); margin: 1px; max-width:540px; min-width:326px; padding:0; width:99.375%; width:-webkit-calc(100% - 2px); width:calc(100% - 2px);"><div style="padding:16px;"> <a href="https://www.instagram.com/p/B7EbX3hnnbu/?utm_source=ig_embed&utm_campaign=loading" style=" background:#FFFFFF; line-height:0; padding:0 0; text-align:center; text-decoration:none; width:100%;" target="_blank"> <div style=" display: flex; flex-direction: row; align-items: center;"> <div style="background-color: #F4F4F4; border-radius: 50%; flex-grow: 0; height: 40px; margin-right: 14px; width: 40px;"></div> <div style="display: flex; flex-direction: column; flex-grow: 1; justify-content: center;"> <div style=" background-color: #F4F4F4; border-radius: 4px; flex-grow: 0; height: 14px; margin-bottom: 6px; width: 100px;"></div> <div style=" background-color: #F4F4F4; border-radius: 4px; flex-grow: 0; height: 14px; width: 60px;"></div></div></div><div style="padding: 19% 0;"></div> <div style="display:block; height:50px; margin:0 auto 12px; width:50px;"><svg width="50px" height="50px" viewBox="0 0 60 60" version="1.1" xmlns="https://www.w3.org/2000/svg" xmlns:xlink="https://www.w3.org/1999/xlink"><g stroke="none" stroke-width="1" fill="none" fill-rule="evenodd"><g transform="translate(-511.000000, -20.000000)" fill="#000000"><g><path d="M556.869,30.41 C554.814,30.41 553.148,32.076 553.148,34.131 C553.148,36.186 554.814,37.852 556.869,37.852 C558.924,37.852 560.59,36.186 560.59,34.131 C560.59,32.076 558.924,30.41 556.869,30.41 M541,60.657 C535.114,60.657 530.342,55.887 530.342,50 C530.342,44.114 535.114,39.342 541,39.342 C546.887,39.342 551.658,44.114 551.658,50 C551.658,55.887 546.887,60.657 541,60.657 M541,33.886 C532.1,33.886 524.886,41.1 524.886,50 C524.886,58.899 532.1,66.113 541,66.113 C549.9,66.113 557.115,58.899 557.115,50 C557.115,41.1 549.9,33.886 541,33.886 M565.378,62.101 C565.244,65.022 564.756,66.606 564.346,67.663 C563.803,69.06 563.154,70.057 562.106,71.106 C561.058,72.155 560.06,72.803 558.662,73.347 C557.607,73.757 556.021,74.244 553.102,74.378 C549.944,74.521 548.997,74.552 541,74.552 C533.003,74.552 532.056,74.521 528.898,74.378 C525.979,74.244 524.393,73.757 523.338,73.347 C521.94,72.803 520.942,72.155 519.894,71.106 C518.846,70.057 518.197,69.06 517.654,67.663 C517.244,66.606 516.755,65.022 516.623,62.101 C516.479,58.943 516.448,57.996 516.448,50 C516.448,42.003 516.479,41.056 516.623,37.899 C516.755,34.978 517.244,33.391 517.654,32.338 C518.197,30.938 518.846,29.942 519.894,28.894 C520.942,27.846 521.94,27.196 523.338,26.654 C524.393,26.244 525.979,25.756 528.898,25.623 C532.057,25.479 533.004,25.448 541,25.448 C548.997,25.448 549.943,25.479 553.102,25.623 C556.021,25.756 557.607,26.244 558.662,26.654 C560.06,27.196 561.058,27.846 562.106,28.894 C563.154,29.942 563.803,30.938 564.346,32.338 C564.756,33.391 565.244,34.978 565.378,37.899 C565.522,41.056 565.552,42.003 565.552,50 C565.552,57.996 565.522,58.943 565.378,62.101 M570.82,37.631 C570.674,34.438 570.167,32.258 569.425,30.349 C568.659,28.377 567.633,26.702 565.965,25.035 C564.297,23.368 562.623,22.342 560.652,21.575 C558.743,20.834 556.562,20.326 553.369,20.18 C550.169,20.033 549.148,20 541,20 C532.853,20 531.831,20.033 528.631,20.18 C525.438,20.326 523.257,20.834 521.349,21.575 C519.376,22.342 517.703,23.368 516.035,25.035 C514.368,26.702 513.342,28.377 512.574,30.349 C511.834,32.258 511.326,34.438 511.181,37.631 C511.035,40.831 511,41.851 511,50 C511,58.147 511.035,59.17 511.181,62.369 C511.326,65.562 511.834,67.743 512.574,69.651 C513.342,71.625 514.368,73.296 516.035,74.965 C517.703,76.634 519.376,77.658 521.349,78.425 C523.257,79.167 525.438,79.673 528.631,79.82 C531.831,79.965 532.853,80.001 541,80.001 C549.148,80.001 550.169,79.965 553.369,79.82 C556.562,79.673 558.743,79.167 560.652,78.425 C562.623,77.658 564.297,76.634 565.965,74.965 C567.633,73.296 568.659,71.625 569.425,69.651 C570.167,67.743 570.674,65.562 570.82,62.369 C570.966,59.17 571,58.147 571,50 C571,41.851 570.966,40.831 570.82,37.631"></path></g></g></g></svg></div><div style="padding-top: 8px;"> <div style=" color:#3897f0; font-family:Arial,sans-serif; font-size:14px; font-style:normal; font-weight:550; line-height:18px;"> View this post on Instagram</div></div><div style="padding: 12.5% 0;"></div> <div style="display: flex; flex-direction: row; margin-bottom: 14px; align-items: center;"><div> <div style="background-color: #F4F4F4; border-radius: 50%; height: 12.5px; width: 12.5px; transform: translateX(0px) translateY(7px);"></div> <div style="background-color: #F4F4F4; height: 12.5px; transform: rotate(-45deg) translateX(3px) translateY(1px); width: 12.5px; flex-grow: 0; margin-right: 14px; margin-left: 2px;"></div> <div style="background-color: #F4F4F4; border-radius: 50%; height: 12.5px; width: 12.5px; transform: translateX(9px) translateY(-18px);"></div></div><div style="margin-left: 8px;"> <div style=" background-color: #F4F4F4; border-radius: 50%; flex-grow: 0; height: 20px; width: 20px;"></div> <div style=" width: 0; height: 0; border-top: 2px solid transparent; border-left: 6px solid #f4f4f4; border-bottom: 2px solid transparent; transform: translateX(16px) translateY(-4px) rotate(30deg)"></div></div><div style="margin-left: auto;"> <div style=" width: 0px; border-top: 8px solid #F4F4F4; border-right: 8px solid transparent; transform: translateY(16px);"></div> <div style=" background-color: #F4F4F4; flex-grow: 0; height: 12px; width: 16px; transform: translateY(-4px);"></div> <div style=" width: 0; height: 0; border-top: 8px solid #F4F4F4; border-left: 8px solid transparent; transform: translateY(-4px) translateX(8px);"></div></div></div> <div style="display: flex; flex-direction: column; flex-grow: 1; justify-content: center; margin-bottom: 24px;"> <div style=" background-color: #F4F4F4; border-radius: 4px; flex-grow: 0; height: 14px; margin-bottom: 6px; width: 224px;"></div> <div style=" background-color: #F4F4F4; border-radius: 4px; flex-grow: 0; height: 14px; width: 144px;"></div></div></a><p style=" color:#c9c8cd; font-family:Arial,sans-serif; font-size:14px; line-height:17px; margin-bottom:0; margin-top:8px; overflow:hidden; padding:8px 0 7px; text-align:center; text-overflow:ellipsis; white-space:nowrap;"><a href="https://www.instagram.com/p/B7EbX3hnnbu/?utm_source=ig_embed&utm_campaign=loading" style=" color:#c9c8cd; font-family:Arial,sans-serif; font-size:14px; font-style:normal; font-weight:normal; line-height:17px; text-decoration:none;" target="_blank">A post shared by @fashionista.couture</a> on <time style=" font-family:Arial,sans-serif; font-size:14px; line-height:17px;" datetime="2020-01-08T18:44:39+00:00">Jan 8, 2020 at 10:44am PST</time></p></div></blockquote> <script async src="//www.instagram.com/embed.js"></script>
  - Twitter
  
  <blockquote class="twitter-tweet"><p lang="en" dir="ltr">A <a href="https://twitter.com/hashtag/SussexRoyal?src=hash&amp;ref_src=twsrc%5Etfw">#SussexRoyal</a> birthday message from Meghan to Harry — and a previously unseen photo from Archie’s christening!<br><br>“Your service to the causes you care so deeply for inspires me every day. You are the best husband and most amazing dad to our son. We love you ❤️Happiest birthday!” <a href="https://t.co/Nghf22E6Sw">pic.twitter.com/Nghf22E6Sw</a></p>&mdash; Omid Scobie (@scobie) <a href="https://twitter.com/scobie/status/1173153935177244672?ref_src=twsrc%5Etfw">September 15, 2019</a></blockquote> <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
  
  - Tiktok
  <blockquote class="tiktok-embed" cite="https://www.tiktok.com/@careadactyl1/video/6893942869250329861" data-video-id="6893942869250329861" style="max-width: 605px;min-width: 325px;" > <section> <a target="_blank" title="@careadactyl1" href="https://www.tiktok.com/@careadactyl1">@careadactyl1</a> <p><a title="playroom" target="_blank" href="https://www.tiktok.com/tag/playroom">#playroom</a> <a title="montessori" target="_blank" href="https://www.tiktok.com/tag/montessori">#montessori</a> <a title="parentsoftiktok" target="_blank" href="https://www.tiktok.com/tag/parentsoftiktok">#parentsoftiktok</a> <a title="parent" target="_blank" href="https://www.tiktok.com/tag/parent">#parent</a> <a title="fail" target="_blank" href="https://www.tiktok.com/tag/fail">#fail</a> <a title="help" target="_blank" href="https://www.tiktok.com/tag/help">#help</a> <a title="failarmy" target="_blank" href="https://www.tiktok.com/tag/failarmy">#failarmy</a> <a title="interiordesigner" target="_blank" href="https://www.tiktok.com/tag/interiordesigner">#interiordesigner</a></p> <a target="_blank" title="♬ original sound - Matthew Rincon" href="https://www.tiktok.com/music/original-sound-6866557456730852102">♬ original sound - Matthew Rincon</a> </section> </blockquote> <script async src="https://www.tiktok.com/embed.js"></script>
  
  ### IFRAME
  - Youtube
  <iframe width="560" height="315" src="https://www.youtube.com/embed/7YPYP0hO8y4" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
  
  - assets
  <iframe src="https://assets.pinterest.com/ext/embed.html?id=449163762808076559" height="550" width="345" frameborder="0" scrolling="no" ></iframe>
  
  - GIPHY
  <iframe src="https://giphy.com/embed/WMswWecHv29P2" width="480" height="480" frameBorder="0" class="giphy-embed" allowFullScreen></iframe><p><a href="https://giphy.com/gifs/WMswWecHv29P2">via GIPHY</a></p>
  
  - BBC
  <iframe width="400" height="500" frameborder="0" src="https://www.bbc.com/news/av/embed/p08dxtfd/52786833"></iframe>
  
  - FB : Post
  <iframe src="https://www.facebook.com/plugins/post.php?href=https%3A%2F%2Fwww.facebook.com%2Fhairdesignbybritney%2Fposts%2F103659755204446&width=500&show_text=true&height=786&appId" width="500" height="786" style="border:none;overflow:hidden" scrolling="no" frameborder="0" allowfullscreen="true" allow="autoplay; clipboard-write; encrypted-media; picture-in-picture; web-share"></iframe>
  
  - FB Video
  
  <iframe src="https://www.facebook.com/plugins/video.php?href=https%3A%2F%2Fwww.facebook.com%2Fredtabletalk%2Fvideos%2F419568041972431%2F&show_text=0&width=560" width="560" height="315" style="border:none;overflow:hidden" scrolling="no" frameborder="0" allowfullscreen="true" allow="autoplay; clipboard-write; encrypted-media; picture-in-picture; web-share" allowFullScreen="true"></iframe>
  
  - Reddit
  <iframe id="reddit-embed" src="https://www.redditmedia.com/r/pics/comments/ijkmd9/tribute_tattoos_for_mom_and_grandmother/?ref_source=embed&amp;ref=share&amp;embed=true" sandbox="allow-scripts allow-same-origin allow-popups" scrolling="no" width="640" height="553"></iframe>
  
  - Linkedin
  <iframe src="https://www.linkedin.com/embed/feed/update/urn:li:share:6787493233900572672" height="910" width="504" frameborder="0" allowfullscreen="" title="Embedded post"></iframe>
  
  ### A Tag
  - Youtube
  <a href="https://www.youtube.com/embed/3CeCIYt9NQQ" class="embedly-card" data-card-width="100%" data-card-controls="0">Embedded content: https://www.youtube.com/embed/3CeCIYt9NQQ</a>
  
  - Twitter
  <a href="https://twitter.com/Frank__Massaro/status/1306161764006948864" class="embedly-card" data-card-width="100%" data-card-controls="0">Embedded content: https://twitter.com/Frank__Massaro/status/1306161764006948864</a>
  
  - Facebook
  <a href="https://www.facebook.com/mamaslatinas/videos/2081111495236962/&show_text=0&width=560" class="embedly-card" data-card-width="100%" data-card-controls="0">Embedded content: https://www.facebook.com/mamaslatinas/videos/2081111495236962/&show_text=0&width=560</a>
  
  - Pinterest
  <a href="https://www.pinterest.com/pin/158822324335196276/" class="embedly-card" data-card-width="100%" data-card-controls="0">Embedded content: https://www.pinterest.com/pin/158822324335196276/</a>
  
  - GIPHY
  
  <a href="https://media.giphy.com/media/QZ1afO5RVJZTOw2FkH/giphy.gif" class="embedly-card" data-card-width="100%" data-card-controls="0">Embedded content: https://media.giphy.com/media/QZ1afO5RVJZTOw2FkH/giphy.gif</a>
  
  ## DIV
  - Schema
  <div itemscope itemtype="https://schema.org/VideoObject"><meta itemprop="uploadDate" content="Wed Nov 18 2020 08:02:49 GMT-0500 (Eastern Standard Time)"/><meta itemprop="name" content="Truth Bomb Moms: I Have A Dragon Child"/><meta itemprop="duration" content="PT3M27.322S" /><meta itemprop="thumbnailUrl" content="https://content.jwplatform.com/thumbs/XUeghygr-1920.jpg"/><meta itemprop="contentUrl" content="https://content.jwplatform.com/videos/XUeghygr-1B3ek2o0.mp4"/><div style="position:relative; overflow:hidden; padding-bottom:56.25%"> <iframe src="https://cdn.jwplayer.com/players/XUeghygr-uSWhbiuC.html" width="100%" height="100%" frameborder="0" scrolling="auto" title="Truth Bomb Moms: I Have A Dragon Child" style="position:absolute;" allowfullscreen></iframe> </div></div>
  
  - Tumblr
  <div class="tumblr-post" data-href="https://embed.tumblr.com/embed/post/wW6bSswFg1kY1-ZT4RORjg/163529719642" data-did="f69d932f7b4858f6ef7fff48c41d97ea35ce726e"><a href="https://tattooingisanart.tumblr.com/post/163529719642/avalontattoo">https://tattooingisanart.tumblr.com/post/163529719642/avalontattoo</a></div>  <script async src="https://assets.tumblr.com/post.js"></script>
    
  ## Normal Blockquote
  <blockquote>Do not put this into EmbedImage</blockquote>
  `,
}
