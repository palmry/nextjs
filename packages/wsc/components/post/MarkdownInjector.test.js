import React from 'react'
import { render } from '@testing-library/react'
import MarkdownInjector, { processMarkdown, createMarkdownPlaceHolder } from './MarkdownInjector'
import { ButtonAction } from '../button/ButtonAction'
import { DATA } from './__fixtures__/MarkdownInjector'

const ButtonActionInjector = jest.fn(index => {
  // insert after first paragraph
  if (index === 0) {
    return {
      after: <ButtonAction text={'OK'} key={`button-${index}`} />,
    }
  }
})

describe('processMarkdown function', () => {
  it('does nothing if markdown is null', () => {
    const components = processMarkdown(null, null)
    expect(components).toBeUndefined()
  })
  it('can inject any components to markdown', () => {
    const components = processMarkdown([ButtonActionInjector], DATA.markdownData)

    // ensure the injector function is called
    expect(ButtonActionInjector).toBeCalled()

    expect(components).toMatchSnapshot()
  })
})

describe('MarkdownInjector', () => {
  it('does nothing if markdown is null', () => {
    const { container } = render(<MarkdownInjector componentInjectors={[ButtonActionInjector]} />)
    expect(container.firstChild).toBeNull()
  })
  it('can inject any components to markdown', () => {
    const { container } = render(
      <MarkdownInjector componentInjectors={[ButtonActionInjector]} markdown={DATA.markdownData} />
    )
    expect(container).toMatchSnapshot()
  })
})

/*
We will enable this once we fixed twitter quote issue.
describe('createMarkdownPlaceHolder', () => {
  it('can replace embed code with EmbedImage component', () => {
    const markdownList = createMarkdownPlaceHolder(DATA.embedCodeData)
    expect(markdownList).toMatchSnapshot()
  })
})*/
