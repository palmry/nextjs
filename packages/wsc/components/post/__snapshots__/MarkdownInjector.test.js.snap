// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkdownInjector can inject any components to markdown 1`] = `
<div>
  <div
    class="ContentMarkdown__ScHTMLContent-sc-1qkx5c8-0 dtgUXI"
  >
    <p>
      <strong>
        1st paragraph starts
      </strong>
      <a
        href="https://www.wayfair.com/daily-sales/way-day"
        rel="noopener"
        target="_blank"
      >
        Wayfair's "Way Day" sale
      </a>
       totally beats them all
    </p>
  </div>
  <button
    class="ButtonAction__Button-pe4a18-0 jiiVhH buttonAction"
  >
    OK
  </button>
  <div
    class="ContentMarkdown__ScHTMLContent-sc-1qkx5c8-0 dtgUXI"
  >
    <p>
      <strong>
        2nd paragraph starts
      </strong>
    </p>
  </div>
</div>
`;

exports[`processMarkdown function can inject any components to markdown 1`] = `
Array [
  MarkdownPlaceholder {
    "key": 0,
    "markdown": "**1st paragraph starts**[Wayfair's \\"Way Day\\" sale](https://www.wayfair.com/daily-sales/way-day) totally beats them all

",
  },
  <React.Fragment>
    <ButtonAction
      text="OK"
    />
  </React.Fragment>,
  MarkdownPlaceholder {
    "key": 2,
    "markdown": "**2nd paragraph starts**
",
  },
]
`;
